### PR TITLE
Add a resolve conflict test in ReplicatorCollectionTest

### PR DESCRIPTION
**Pull request by pasin**
* Added a resolve conflict test in ReplicatorCollectionTest.

* Added setConflictResolver() to allow to set a conflict resolver function which is similar to ConflictResolver function in CBL.

### CODE REVIEW

**Review by jianminzhao — APPROVED** [review]

File: Replicator/tests/ReplicatorCollectionTest.cc [hunk 1]
```diff
@@ -147,6 +148,103 @@ class ReplicatorCollectionTest : public ReplicatorLoopbackTest {
...
+        if (!resolver) {
+            _conflictHandler = nullptr;
+            return;
+        }
+        
+        assert(activeDatabase);
+        
+        c4::ref<C4Database> resolvDB = c4db_openAgain(activeDatabase, nullptr);
+        REQUIRE(resolvDB);
+        auto& conflictHandlerRunning = _conflictHandlerRunning;
```
[the remark this answers is not in the document]
Comment (pasin) [reply Replicator/tests/ReplicatorCollectionTest.cc:178]: It is used in ReplicatorLoopbackTest.hh's _checkStopWhenIdle() to delay the stop if the conflict resolution is active.

https://github.com/couchbase/couchbase-lite-core/blob/master/Replicator/tests/ReplicatorLoopbackTest.hh#L196

File: Replicator/tests/ReplicatorCollectionTest.cc [hunk 2]
```diff
@@ -147,6 +148,103 @@ class ReplicatorCollectionTest : public ReplicatorLoopbackTest {
...
+    class ResolvedDocument {
+    public:
+        ResolvedDocument()                      =default;   // Resolved as a deleted doc
+        ResolvedDocument(C4Document* doc)       :_doc(c4doc_retain(doc)) { }
+        ResolvedDocument(FLDict mergedProps)    :_mergedProps(mergedProps) { }
+        
+        C4Document* doc()                       {return _doc;}
+        FLDict mergedProps()                    {return _mergedProps;}
+    private:
+        c4::ref<C4Document> _doc                {nullptr};
```
[the remark this answers is not in the document]
Comment (pasin) [reply Replicator/tests/ReplicatorCollectionTest.cc:160]: Will remove.

File: Replicator/tests/ReplicatorCollectionTest.cc [hunk 3]
```diff
@@ -147,6 +148,103 @@ class ReplicatorCollectionTest : public ReplicatorLoopbackTest {
...
+        RetainedDict _mergedProps               {nullptr};
```
[the remark this answers is not in the document]
Comment (pasin) [reply Replicator/tests/ReplicatorCollectionTest.cc:161]: Will remove.

**Comment by pasin** [comment]
Test failure (Rapid Restarts) on Mac is not related to this PR.

**Review by borrrden — APPROVED** [review]

**Pull request merged**